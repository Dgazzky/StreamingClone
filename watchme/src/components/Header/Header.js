import React from 'react'
import styles from './Header.module.scss'
import { STRINGS } from '../../constants/strings'

const Header = () => {
    return (
        <div className={styles.header}>
            <div className={styles.circle}> </div>
            <div className={styles.title}> {STRINGS['label']}</div>
            <div className={styles.divider}> </div>
        </div>
    )
}
export default Header
