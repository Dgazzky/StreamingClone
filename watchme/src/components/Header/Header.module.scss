@import '../../constants/colors';
.header {
    width: 100%;
    height: 73px;
    background-color: $color-blue-8;
    display: flex;
    flex-direction: row;
    align-items: center;
}
.circle {
    border: 2px solid $color-voilet-200;
    border-radius: 50%;
    width: 17px;
    height: 17px;
    margin-left: 27px;
}
.title {
    font-family: 'Montserrat';
    color: $color-neutral-000;
    font-size: 14px;
    margin-left: 12px;
}
.divider {
    height: 35px;
    width: 2px;
    background-color: $color-gray-400;
    margin-left: 31px;
}
