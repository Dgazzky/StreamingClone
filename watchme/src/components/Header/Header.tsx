import React from 'react'
import { Switch } from 'antd'
import styles from './Header.module.scss'
import { STRINGS } from '../../constants/strings'
import Search from './Search'
import { useSelector } from 'react-redux'
import { MoviesSelectors } from '../../redux/movies'

interface HeaderProps {
    onSearch: (value: string) => void
    onFavoritesView: () => void
}
const Header = ({ onSearch, onFavoritesView }: HeaderProps) => {
    const handleChange = () => {
        document.body.classList.toggle('dark-theme')
        document.body.classList.toggle('light-theme')
    }

    const isFavoritesMode = useSelector(MoviesSelectors.getMode)
    return (
        <div className={styles.header}>
            <div className={styles.circle}> </div>
            <div className={styles.title}> {STRINGS['label']}</div>
            <div className={styles.divider}> </div>
            <div className={styles.favorites} onClick={onFavoritesView}>
                {isFavoritesMode
                    ? STRINGS['view.all']
                    : STRINGS['view.favorites']}
            </div>
            <div className={styles.theme}>
                <Switch
                    onChange={handleChange}
                    unCheckedChildren={'Dark theme'}
                    checkedChildren={'Light theme'}
                />
            </div>
            <div className={styles.search}>
                <Search onSearch={onSearch} />
            </div>
        </div>
    )
}
export default Header
