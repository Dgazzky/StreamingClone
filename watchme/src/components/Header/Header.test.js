import React from 'react'

import renderer from 'react-test-renderer'

import Header from './Header'

describe('Search', function () {
    it('renders correctly', () => {
        const component = renderer.create(<Header />).toJSON()
        expect(component).toMatchSnapshot()
    })
})
