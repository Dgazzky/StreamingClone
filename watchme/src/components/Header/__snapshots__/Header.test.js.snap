// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="header"
>
  <div
    className="circle"
  >
     
  </div>
  <div
    className="title"
  >
     
    Recruitment task
  </div>
  <div
    className="divider"
  >
     
  </div>
</div>
`;
